canlink/mcap.go

```diff
@@ -0,0 +1,151 @@
+package canlink
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/foxglove/mcap/go/mcap"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+)
+
+type Mcap struct {
+	l               *zap.Logger
+	file            *os.File
+	createdChannels map[string]uint16
+	client          *CANClient
+	currIndex       uint16
```

Comment:
Does this need to be a field in Mcap type?

Reply:
yeah bc it keeps track of next index for a new channel